contributing.md

```diff
@@ -0,0 +1,235 @@
+file. For R see [R Packages](https://r-pkgs.org/). For other languages check for the
+recommended approach or ask for advice.
+
+### Testing
+
+Ideally codes should contain
+[some tests](https://the-turing-way.netlify.app/reproducible-research/testing)
+to ensure that they work correctly.
+We realise that this may not be necessary in all cases and codes may well have been
+written without testing.
+If you would like help with adding tests to your code this please ask for advice - we
+are happy to support you.
+
+### Continuous Integration
+
+[GitHub Actions](https://docs.github.com/en/actions) can be used to run checks on
+any code that is added to a repository.
+For example, ensuring that it meets certain style, linting, documentation guidelines,
+or to auto-generate documentation and/or webpages.
+We are happy to support you in doing this.
+
+
+## Adding code
+
+### Initial upload
+
+After requesting to add a new code to GO-SHIP we will create a GitHub repository with
+the bare minimum contents.
+To populate this with your code you should navigate into the folder in which it is
+installed on your computer and then run, from the terminal:
+```bash
+git init
+git remote add origin git@github.com:GO-SHIP-Oceanography/<your-go-ship-repo>.git
+git pull origin main
+```
+to pull any base documents (e.g. license).  
+_Note: This assumes you are accessing a repository via ssh (advised). If you prefer to
+access via https you can use
+`git remote add origin https://github.com/GO-SHIP-Oceanography/<your-go-ship-repo>.git`
+and will be prompted for your GitHub username and password each time you access._
+
+Following this you can add files using `[git add](https://github.com/git-guides/git-add)`,
+commit them using `[git commit](https://github.com/git-guides/git-commit)` and push them
+to the repository using `[git push](https://github.com/git-guides/git-push)` as usual.
+The [Turing Way guide](https://the-turing-way.netlify.app/reproducible-research/vcs/vcs-git)
+is a useful guide for those new to these ideas.
+
+We suggest that you perform your initial upload from a branch, allowing the code to be
+reviewed before it is contributed to the main branch, and checked to ensure that it
+works and is reproducible.
+To do this run:
+```bash
+git checkout -b first-commit
+```
+before any `git add` or `git commit` commands.
+Then, when you are ready, run:
+```bash
+git push origin first-commit
+```
+to upload the code to GitHub.
+
+You can then open a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)
+from the repository webpage on GitHub to ask to add your code to the main branch.
+
+If you are unsure about any part of this process we are happy to guide you through it.
+
+### Further developments
+
+Once you have performed the initial upload of the code to the main branch of your
```

Comment:
remove "to the main branch"...? we suggested the initial upload is done on a first-commit branch, right?

Reply:
Done